data-prepper-core/src/main/java/org/opensearch/dataprepper/DataPrepper.java

```diff
@@ -138,8 +139,8 @@ public Map<String, Pipeline> getTransformationPipelines() {
         return transformationPipelines;
     }
 
-    public void registerShutdownHandler(final DataPrepperShutdownListener shutdownListener) {
-        this.shutdownListener = shutdownListener;
+    public void registerShutdownHandlers(final List<DataPrepperShutdownListener> shutdownListeners) {
```

Comment:
Make this take in just a single item, just as it did before. It helps keep the client code cleaner.